hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/protocol/ScmBlockLocationProtocolServerSideTranslatorPB.java

```diff
@@ -104,7 +104,7 @@ public ScmBlockLocationProtocolServerSideTranslatorPB(
   @Override
   public SCMBlockLocationResponse send(RpcController controller,
       SCMBlockLocationRequest request) throws ServiceException {
-    if (!scm.getScmContext().isLeader()) {
+    if (!scm.checkLeader()) {
```

Comment:
How about move the `StorageContainerManager#checkLeader` into `SCMContext`? The latter is used to hold all kinds of global state of SCM. We can handle this in a new jira.

Reply:
Yes sure this change is not related to this.